lib/Doctrine/ORM/Mapping/Builder/ClassMetadataBuilder.php

```diff
@@ -220,10 +220,11 @@ public function setSingleTableInheritance()
      * @param string $type
      * @param int    $length
      * @psalm-param class-string<BackedEnum>|null $enumType
+     * @psalm-param array<string, mixed> $options
      *
      * @return $this
      */
-    public function setDiscriminatorColumn($name, $type = 'string', $length = 255, ?string $columnDefinition = null, ?string $enumType = null)
+    public function setDiscriminatorColumn($name, $type = 'string', $length = 255, ?string $columnDefinition = null, ?string $enumType = null, array $options = [])
```

Comment:
Technically a breaking change for extending classes, but we didn't frown upon a similar change in https://github.com/doctrine/orm/pull/10288 , so maybe we should just make the class as final on 3.0.x.